src/db/oracledb/jobs-dao.js

```diff
@@ -37,23 +120,316 @@ const getJobs = async (internalId, query) => {
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const updateLaborChangeJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, ['laborDistribution']);
+
+  const { outBinds: { result } } = await connection.execute(contrib.updateLaborChangeJob(), binds);
+  return result;
+};
+
+/**
+ * Updates student job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const studentUpdateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, studentBinds);
+
+  const result = await connection.execute(contrib.studentUpdateJob(binds), binds);
+  return result.outBinds.result;
+};
+
+/**
+ * Creates student job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const studentCreateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, studentBinds);
+
+  const { outBinds: { result } } = await connection.execute(contrib.studentCreateJob(binds), binds);
+  return result;
+};
+
+/**
+ * Creates graduate job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const graduateCreateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib.graduateCreateJob(binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Updates graduate job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const graduateUpdateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib.graduateUpdateJob(binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Determines if changeReasonCode is valid
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} changeReasonCode Change reason code from request body
+ * @returns {boolean} True if changeReasonCode is valid
+ */
+const isValidChangeReasonCode = async (connection, changeReasonCode) => {
+  const { rows } = await connection.execute(
+    contrib.validateChangeReasonCode(),
+    { changeReasonCode },
+  );
+  return rows[0].count > 0;
+};
+
+/**
+ * Determines which Epaf to execute based on fields in body
+ *
+ * @param {boolean} update True if updating record. False if creating record
```

Comment:
Just my two cents, I think using an enum string as controller (`update` and `create`) like is clearer than using boolean here. We can also easily extend in the future if there's a thrid different case we need to handle.